pages/home.py

```diff
@@ -413,6 +499,250 @@ def functional_narrative(df):
     return text
 
 
+def subnational_spending_narrative(
+    df_spending,
+    df_poverty,
+    top_n=3,
+    exp_thresh=0.5,
+    per_capita_thresh=1000,
+    corr_thresholds=(0.3, 0.7),
+):
+    total_expenditure = (
+        df_spending.groupby("adm1_name")["expenditure"]
+        .sum()
+        .sort_values(ascending=False)
+    )
+    top_n_total = total_expenditure.head(top_n)
+    total_expenditure_sum = total_expenditure.sum()
+    top_n_percentage = top_n_total.sum() / total_expenditure_sum
+    per_capita_expenditure = df_spending.groupby("adm1_name")[
+        "per_capita_expenditure"
+    ].mean()
+    per_capita_range = per_capita_expenditure.max() - per_capita_expenditure.min()
+    per_capita_median = per_capita_expenditure.median()
+    if not df_poverty.empty:
+        poverty_rates = df_poverty.groupby("region_name")["poor215"].mean()
+        correlation = per_capita_expenditure.corr(poverty_rates)
+
```

Comment:
Let's add handling for none values.
<img width="1357" alt="Screenshot 2024-08-29 at 4 04 08 PM" src="https://github.com/user-attachments/assets/66d47dd3-4a08-4148-8344-73a9d4398483">